tools/ops/azure/container-host/aci/aci-template.json

```diff
@@ -0,0 +1,115 @@
+            "defaultValue": "[concat('custodian-', parameters('azure_subscription_id'))]",
+            "type": "string"
+        },
+        "location": {
+            "defaultValue": "[resourceGroup().location]",
+            "type": "string"
+        },
+        "image_repository": {
+            "defaultValue": "cloudcustodian/c7n",
+            "type": "string"
+        },
+        "image_tag": {
+            "defaultValue": "latest",
+            "type": "string"
+        },
+        "restart_policy": {
+            "type": "string",
+            "allowedValues": [
+                "Always",
+                "OnFailure",
+                "Never"
+            ],
+            "defaultValue": "Always"
+        },
+        "user_assigned_identity_name": {
+            "type": "string"
+        },
+        "azure_event_queue_name": {
+            "type": "string",
+            "defaultValue": "[concat('custodian-', parameters('azure_subscription_id'))]"
+        },
+        "azure_subscription_id": {
+            "type": "string",
+            "defaultValue": "[subscription().subscriptionId]"
+        },
+        "azure_container_storage": {
+            "type": "string"
+        },
+        "azure_event_queue_resource_id": {
+            "type": "string"
+        }
+    },
+    "variables": {
+        "user_assigned_identity_resource_id": "[resourceId('Microsoft.ManagedIdentity/userAssignedIdentities/', parameters('user_assigned_identity_name'))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.ContainerInstance/containerGroups",
+            "apiVersion": "2018-10-01",
+            "name": "[parameters('aci_name')]",
+            "location": "[parameters('location')]",
+            "identity": {
+                "type": "UserAssigned",
+                "userAssignedIdentities": {
+                    "[variables('user_assigned_identity_resource_id')]": {}
+                }
+            },
+            "properties": {
+                "containers": [
+                    {
+                        "name": "[parameters('aci_name')]",
+                        "properties": {
+                            "image": "[concat(parameters('image_repository'), ':', parameters('image_tag'))]",
+                            "command": [
+                                "/usr/local/bin/python3",
+                                "-m",
+                                "c7n_azure.container_host.host"
+                            ],
+                            "ports": [],
+                            "environmentVariables": [
+                                {
+                                    "name": "AZURE_USE_MSI",
+                                    "value": "1"
+                                },
+                                {
+                                    "name": "AZURE_CLIENT_ID",
+                                    "value": "[reference(variables('user_assigned_identity_resource_id'), '2018-11-30').clientId]"
```

Comment:
Oh you made this work, awesome

Reply:
yeah, the only tricky part was that it will only reference a resource outside of the template if you provide the api version argument. I didn't really see that documented very well